src/lsp/MessageIO.ts

```diff
@@ -129,6 +143,15 @@ export class MessageIOReader extends AbstractMessageReader implements MessageRea
 			this.nextMessageLength = -1;
 			this.messageToken++;
 			var json = JSON.parse(msg);
+			if (json && json.result && json.result.capabilities) {
+				let dataDirectory = json.result.capabilities.dataDirectory
+				new extension.DocContent(dataDirectory)
```

Comment:
Is it possible some fs error thrown from DocContent constructor？  Are they handled somewhere else or we should handle it here?

Reply:
It will be handled here by this if condition since json.result will be null in that case